### ISSUE TITLE
Tiller image not available

### ISSUE THREAD

**Issue by dotariel** [helm/helm#3770]
Output of `helm version`:
```
Client: &version.Version{SemVer:"v2.8+unreleased", GitCommit:"6b83a4125b94f8454bbc8e34f4c727238b2e1082", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.8.2", GitCommit:"a80231648a1473929271764b920a8e346f6de844", GitTreeState:"clean"}
```

Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.2", GitCommit:"5fa2db2bd46ac79e5e00a4e6ed24191080aa463b", GitTreeState:"clean", BuildDate:"2018-01-18T21:12:46Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"", Minor:"", GitVersion:"v1.9.0", GitCommit:"925c127ec6b946659ad0fd596fa959be43f0cc05", GitTreeState:"clean", BuildDate:"2018-01-26T19:04:38Z", GoVersion:"go1.9.1", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
Minikube

---
`helm init` is defaulting to `gcr.io/kubernetes-helm/tiller:v2.8` for the `tiller-deploy` deployment, but that image tag does not exist in the container registry. The available tags are: 
* v2.8.2
* v2.8.2-rc1
* v2.8.1
* v2.8.0
* v2.8.0-rc.1

Temporary workaround is to override the tiller image version during init:

```
helm init --tiller-image gcr.io/kubernetes-helm/tiller:v2.8.2
```

**Comment by bacongobbler** [comment]
Because you're using an unofficial release of Helm, you'll need to go through the steps provided in the documentation around [installing a canary version of tiller](https://docs.helm.sh/using_helm/#installing-tiller-canary-builds), which is:

```
$ helm init --canary-image
```

If you want to test an official release of Tiller with a dev release of Helm, you need to do so with

```
$ helm init --tiller-image gcr.io/kubernetes-helm/tiller:v2.8.2
```

As you already mentioned in your workaround.

Perhaps what we should do is make "unofficial" helm releases ask you to set these feature flags, or just install the canary release by default. I'm more in favour of the former because you may be using an older dev release of helm so the canary image may not work in that situation.

This would be an easy first enhancement if anyone wants to work on this!

**Comment by bacongobbler** [comment]
resolved in #3781
